lib/offline/storage.js

```diff
@@ -1334,10 +1334,19 @@ shaka.offline.Storage = class {
       if (!toDownload.has(pendingSegmentRefId)) {
         const estimateId = downloader.addDownloadEstimate(
             estimator.getSegmentEstimate(stream.id, segment));
+        let groupId = stream.id;
+        const numberOfParallelSegmentsByStream =
+            config.offline.numberOfParallelSegmentsByStream;
+        if (numberOfParallelSegmentsByStream > 1) {
+          // Note: 1000 is a random number to not coincide with other groups
```

Comment:
nit: It's not random, it's arbitrary.  And you should note that 1000 is safe so long as there are fewer than 1000 actual streams.